### ISSUE TITLE
Black screen on wake from sleep

### ISSUE THREAD

**Issue by Saad-LUMS** [hackintosh-stuff/XPS9350-macOS#14]
Hi, installed on xps 13 9350 with 256 gb pm951 nvme ssd, i5 6200u, hd 520, 8 gb ram, disabled wifi card in bios.

Everything works perfectly apart from black screen on waking from sleep. The backlight is on, grey, and I can press keys to get notification sound but display does not show UI or anything other than grey, black.

Sometimes it tries to turn on with instant flashes of UI but nothing, goes to black.
For now I've disabled both laptop and display sleep as both cause issues but it's not a long term solution. 
Any advice?

**Comment by Saad-LUMS** [comment]
There is a related but weird issue where if I lock the screen (only happens on lock screen, but not on password input phase on cold boot or restart) and press Escape button, the display turns off and won't turn on again. All of the other symptoms as above continue.

But here if I press the Delete key, the display turns grey, but does not show anything, with repeated pressings of Del and Esc this pattern continues on and on. Everything else works, other keys produce the notification sounds.

**Comment by Xerz** [comment]
I don't have the fix, but "workaround" for me is to use the lid only. I've been on 14.1 for about a year now, no issues with sleep when i close the lid to go to sleep and open the lid to wake up (but if i go to sleep from menu or by timeout, then it can or cannot behave as you described).

**Comment by Saad-LUMS** [comment]
Interesting. I did not try that. Will do.
In the mean time, I used Syscl's deploy.sh and it solved that issue but then the trackpad doesn't work.
Will update on final config.

**Comment by Saad-LUMS** [comment]
No go with lid wake tweak. I am on 10.14.6 btw.

**Comment by Saad-LUMS** [comment]
@maz-1 Please look into this. Your Clover is near perfect, apart from just this 1 issue (The other Authentication Delay one I identified, is not that big and can be lived with).  It gives the fastest boot times, with amazing trackpad features and accuracy and nearly everything works. I am currently using it as my daily driver and typing this on it.
 
I tried Syscl's, Tlefko's and so many other's files. They all had weird issues like poor trackpad implementations and whatnot, none were nearly as complete and as easy to implement as yours.

We could have a perfect hackintosh XPS if this is fixed.

I am happy to provide more info to help but I am not a coder and cannot solve this issue myself.

**Comment by carking** [comment]
> @maz-1 Please look into this. Your Clover is near perfect, apart from just this 1 issue (The other Authentication Delay one I identified, is not that big and can be lived with). It gives the fastest boot times, with amazing trackpad features and accuracy and nearly everything works. I am currently using it as my daily driver and typing this on it.
> 
> I tried Syscl's, Tlefko's and so many other's files. They all had weird issues like poor trackpad implementations and whatnot, none were nearly as complete and as easy to implement as yours.
> 
> We could have a perfect hackintosh XPS if this is fixed.
> 
> I am happy to provide more info to help but I am not a coder and cannot solve this issue myself.

Hi, Do you have any battery drain issue?? I am using another efi right now, my screen is 3k model, the battery drops 1% every two minutes, is that normal? The bottom cover will also become hot (the fan will not work actively)

**Comment by Saad-LUMS** [comment]
@carking Hey. I have an old battery. Backup time on Windows was pretty bad these days. 2 to 3 hours. Getting similar results on MacOS Mojave.

My fan works, I think. Haven't listened to it but the laptop is not getting any hotter than usual.

**Comment by Saad-LUMS** [comment]
I have used the proper Power Management profile provided by @maz-1 for my 6200U though.

**Comment by carking** [comment]
> I have used the proper Power Management profile provided by @maz-1 for my 6200U though.

So is this EFI for your daily use? I'm going downgrage from catalina to mojave to try fix my battery problem.

Do I need to manually import this cpupm file into the kext folder? The original clover folder does not appear to contain the Power Management profile.

**Comment by Saad-LUMS** [comment]
Yeah. I am daily driving  it. Apart from sleep, it's all good. I just use a screensaver all the time.

The detailed full Readme in this github is very helpful. The profile kexts are in Kexts folder outside of Clover in the main directory. 3 ways of doing this. Follow what you like. Read the instructions. 

How did you get Catalina working on 9350? Your specs and Clover used?

**Comment by carking** [comment]
> Yeah. I am daily driving it. Apart from sleep, it's all good. I just use a screensaver all the time.
> 
> The detailed full Readme in this github is very helpful. The profile kexts are in Kexts folder outside of Clover in the main directory. 3 ways of doing this. Follow what you like. Read the instructions.
> 
> How did you get Catalina working on 9350? Your specs and Clover used?

I found the EFI for 10.15 in a Chinese forum and everything worked fine except for my battery drain problem and my configuration was 6560U with 3K display.

**Comment by Saad-LUMS** [comment]
Cool. Can you link the EFI folder please? Might give it a try.

Haven't found a good EFI for Catalina in major forums for 9350 yet. All are for 9360 sadly. But I am happy with Mojave for now until Catalina reaches its .6  final iteration.

**Comment by carking** [comment]
> Cool. Can you link the EFI folder please? Might give it a try.
> 
> Haven't found a good EFI for Catalina in major forums for 9350 yet. All are for 9360 sadly. But I am happy with Mojave for now until Catalina reaches its .6 final iteration.

[EFI10.15.3摄像头修复.zip](https://github.com/hackintosh-stuff/XPS9350-macOS/files/4335640/EFI10.15.3.zip)
I don't know if it's the EFI issue or my laptop issue, but my hackintosh consumes more power than my Windows.

**Comment by Saad-LUMS** [comment]
Hmm.  Try this EFI for Mojave and provide feedback.

**Comment by carking** [comment]
> Hmm. Try this EFI for Mojave and provide feedback.

Did you try it? Also, what is the problem using syscl's EFI on Mojave?

**Comment by Saad-LUMS** [comment]
I'm not going to mess around with my setup for a while now since it's pretty good. I was saying you try using this GitHub with Mojave and let us know if your battery issue is resolved. Battery life on MacOS will always be less than on Windows since for our hardware, macOS is not that optimized.

Also, you could disable touchscreen and SD card reader for extra battery life.

With Syscl's EFI, after running deploy.sh, the touchpad does not work, and the boot time is really long with multiple looping entries of VerbStub being shown in verbose mode related to the PS2touchpad controller which doesn't work.

This EFI has amazing, near mac-like touchpad gesture support with VoodooI2C kext.

**Comment by tlefko** [comment]
> @maz-1 Please look into this. Your Clover is near perfect, apart from just this 1 issue (The other Authentication Delay one I identified, is not that big and can be lived with). It gives the fastest boot times, with amazing trackpad features and accuracy and nearly everything works. I am currently using it as my daily driver and typing this on it.
> 
> I tried Syscl's, Tlefko's and so many other's files. They all had weird issues like poor trackpad implementations and whatnot, none were nearly as complete and as easy to implement as yours.
> 
> We could have a perfect hackintosh XPS if this is fixed.
> 
> I am happy to provide more info to help but I am not a coder and cannot solve this issue myself.

Hey @Saad-LUMS, would you mind giving me some input on my files? Like what specifically has been an issue for you --- I also have the lidwake bug and for the life of me cannot figure out how to fix it.

**Comment by Saad-LUMS** [comment]
Hey @tlefko. I just looked at your reply on my insanelymac post in your Mojave thread and posted a response (solution) there too.
Your guide was actually what gave the me the courage and motivation to go through with this (wipe Win 10 completely). It was very simple to follow along especially since you had the same model as mine.
As for the issues, with your EFI, after the first stage of installation (drive format), the laptop reboots for the second phase but where it's supposed to enter the black apple logo loading bar, it reboots over and over...
Also, since you said you use VoodooPS2 for trackpad, I do not like that very much as the gestures are just keys mapped to the trackpad which is not as smooth and mac-like as VoodooI2C in this EFI.
I see that you have a Catalina github now too. How did you fix the lidwake bug in that? Will test that once Catalina hits the final version. For now, I'm happy daily driving Mojave.

You're doing an awesome job man. Don't stop supporting our XPS which is a whole lot better than any Macbook out there.

**Comment by tlefko** [comment]
@Saad-LUMS yeah the Catalina is definitely worth a test for a second partition. I did upgrade it to I2C like you suggested and based on the input I got. As for the lid wake, I truthfully am not entirely sure what fixed it, it just worked? So I can't really complain. I'd recommend Catalina over Mojave actually for stability the battery has been better in my experience and performance. Thanks for the support!

**Comment by Saad-LUMS** [comment]
@tlefko now you're tempting me! (especially with sleep being fixed). I will give it a spin in a few days. Have a lot of assignments and deadlines atm. Will report my progress back here and at your github thread.

**Comment by tlefko** [comment]
@Saad-LUMS Hahaha, sounds great for sure let me know when you give it a try, guide is even easier to follow. (ZERO post install steps)

**Comment by Saad-LUMS** [comment]
@tlefko you said in your Catalina guide that USB devices eject upon sleep. Perhaps try giving the USBFix patch in this EFI a go which supposedly fixes that. I can't come back from sleep (hehe) so don't know if it works though.

**Comment by tlefko** [comment]
@Saad-LUMS I'll give it a shot, for some reason running the deploy for this made my computer unbootable. I'll try to manually install it

**Comment by tlefko** [comment]
Yeah running the deploy caused a lot of slowdown issues and just some overall strange problems, especially with preformance. It also broke my internet . I'm going to work on this and figure out how to properly strip it down to the essentials. For now usb ejecting on sleep isn't worth the risks until I find a stable solution

**Comment by Saad-LUMS** [comment]
Yeah, don't run the complete deploy, which adds unnecessary steps. Just the USBFix script individually. That's how I do it.

**Comment by tlefko** [comment]
> Yeah, don't run the complete deploy, which adds unnecessary steps. Just the USBFix script individually. That's how I do it.

Is there a reverse Deploy... Otherwise I need to redo the installation

**Comment by Saad-LUMS** [comment]
Damn! Sorry for the mess.

**Comment by tlefko** [comment]
Than I'll be developing my own post install steps

**Comment by tlefko** [comment]
> Damn! Sorry for the mess.

I'll work it out and add it to my repo once I fix this lol, it's part of the process

**Comment by Saad-LUMS** [comment]
The full deploy also installs kexts for Wifi in SLE, which may be causing issues since you already have kexts in Clover/Other I guess. Hence the internet breakage you mentioned and unbootable state.

Best of luck!

**Comment by maz-1** [comment]
I dont have i5 6200u so can't debug, maybe you can play around with whatevergreen to see if there are better results
